use_cases/eluc/data/eluc_data.py

```diff
@@ -0,0 +1,195 @@
+import warnings
+import os
+import xarray as xr
+import regionmask
+import pandas as pd
+
+from datasets import load_dataset, Dataset
+
+from unileaf_util.framework.transformers.data_encoder import DataEncoder
+
+from data import constants
+
+class ELUCData():
+    """
+    Object to automatically handle the processing of ELUC data.
+    Load with import_data() then process into a df with da_to_df().
+    Maintains train and test dataframes, encoder for data, and encoded versions of train and test.
+    """
+    
+    def __init__(self, path: str, update_path=None, start_year=1851, test_year=2012, end_year=2022, countries=None):
+        """
+        If update_path is given, load raw data the old way using 2 files that are merged.
+        Otherwise, path is taken to be a huggingface repo and we load the data from there.
+        """
+        assert start_year < test_year and test_year < end_year
+
+        if update_path:
+            raw = self.import_data(path, update_path)
+            df = self.da_to_df(raw, start_year, end_year, countries)
+
+        else:
+            df = self.hf_to_df(path)
+            if countries:
+                df = self.subset_countries(df, countries)
+
+        self.train_df = df.loc[start_year:test_year]
+        self.test_df = df.loc[test_year:end_year]
+        
+        self.encoder = DataEncoder(self.get_fields(), constants.CAO_MAPPING)
+        self.encoded_train_df = None
+        self.encoded_test_df = None
+
+    def subset_countries(self, df, countries):
+        """
+        Subsets dataframe by country list
+        """
+        idx = constants.COUNTRIES_DF[constants.COUNTRIES_DF["abbrevs"].isin(countries)].index.values
+        return df[df["country"].isin(idx)].copy()
+
+    def hf_to_df(self, hf_repo):
+        """
+        Loads dataset from huggingface, converts to pandas, then sets indices appropriately to time/lat/lon.
+        Keep old time/lat/lon columns so we can use them as features later.
+        """
+        ds = load_dataset(hf_repo)["train"]
```

Comment:
As of now, the entire dataset is labeled as "train" on huggingface, we can modify this later.